README.md

```diff
@@ -325,17 +325,31 @@ When you create one, the SPARQL Algebra expression that is passed will be transf
 
 ### Type System
 
-[See functions/Core.ts](./lib/functions/Core.ts)
-
-The type system is tailored for doing (supposedly) quick evaluation of overloaded functions. A function object consists of a map of argument types to a concrete function implementation for those argument types.
-
-When a function object is called with some functions, it looks up a concrete implementation. If we can not find one, we consider the argument of invalid types.
-
-Since many derived types exist, we also associate every literal with it's primitive datatype when constructing a literal. This handles **subtype substitution**, as we define allowed types in function of these primitives types. Note: the derived type is not maintained after an operation on the term, since I found no functions for which this was relevant.
-
-**Type promotion** is handled in a couple of ways. Firstly, a bit like C++ vtables, if we can not find a implementation for the concrete (primitive) types, we try to find an implementation for the term-types of all the arguments, if that fails, we look for an implementation on generic terms.
-
-We also handle type promotion by explicitly coding for it. This is done in the arithmetic functions `+, -, *, /`.
```

Comment:
@wschella, the things that I removed here, what do they mean? 
Anyway, I should take a look at this!

Reply:
Resolved in suggestion.

Reply:
I unresolved this since It's still imported for me to check the functions and I'll use this as a reminder :)

Reply:
This is done now. I checked all functions.